packages/flutter_tools/lib/src/build_system/targets/common.dart

```diff
@@ -244,23 +245,25 @@ abstract class AotElfBase extends Target {
       reportTimings: false,
       fileSystem: environment.fileSystem,
       logger: environment.logger,
-      xcode: globals.xcode,
+      xcode: globals.xcode!,
```

Comment:
is this ensured to exist?

Reply:
In production, yes, it's set up in the context.  
https://github.com/flutter/flutter/blob/5a85c0d81c2a1c101c1ebc38af15922f72ff1e30/packages/flutter_tools/lib/src/context_runner.dart#L346
I have seen `Google testing` failures when the tool tests in g3 are missing `!` global objects set up in its test context.  It's easy to g3fix them though.